meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register

```diff
@@ -0,0 +1,258 @@
+            "--login",
+            f"--write-object={objfile}",
+            f"--type={type}",
+            f"--id={slot}",
+            f"--label={label}",
+        ]
+        run_quietly(args)
+
+    def has_labels(self, labels: List[str]) -> bool:
+        args = ["pkcs11-tool", f"--module={self.LIB}", f"--pin={PIN}", "--list-objects"]
+        out = subprocess.check_output(args)
+        missing = {x: 1 for x in labels}
+        for line in out.decode().splitlines():
+            for l in labels:
+                if line.find(l) != -1:
+                    try:
+                        del missing[l]
+                    except KeyError:
+                        pass  # already removed
+        return len(missing) == 0
+
+    @classmethod
+    def _is_initialized(cls) -> bool:
+        args = ["pkcs11-tool", f"--module={cls.LIB}", f"--pin={PIN}", "--list-objects"]
+        try:
+            subprocess.check_output(args, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError:
+            return False
+        return True
+
+    @classmethod
+    def _initialize(cls):
+        args = ["pkcs11-tool", f"--module={cls.LIB}", f"--so-pin={SO_PIN}"]
+        subprocess.check_call(args + ["--init-token", "--label", "aktualizr"])
+        subprocess.check_call(args + ["--init-pin", f"--pin={PIN}"])
+
+    @classmethod
+    def get_initialized(cls) -> "Pkcs11":
+        if cls._is_initialized():
+            log.info("Pkcs11 slots already initialized")
+        else:
+            log.info("Initializing pkcs11 slots...")
+            cls._initialize()
+        return cls()
+
+
+class SssCli:
+    @contextmanager
+    def i2c_session(*args, **kwds):
+        log.info("Connecting to ssscli...")
+        subprocess.check_call(["ssscli", "connect", "se05x", "t1oi2c", "none"])
+        try:
+            yield
+        finally:
+            log.info("Disconnecting from ssscli")
+            subprocess.check_call(["ssscli", "disconnect"])
+
+    @classmethod
+    def read_id_list(cls) -> List[str]:
+        with cls.i2c_session():
+            out = subprocess.check_output(["ssscli", "se05x", "readidlist"])
+            lines = out.decode().splitlines()
+            # lines are like: Key-Id: 0Xf0000003   BINARY .....
+            return [x.split(" ")[1] for x in lines if x]
+
+    @classmethod
+    def get_cert(cls, keyid: str, certfile: str):
+        with cls.i2c_session():
+            subprocess.check_call(["ssscli", "get", "cert", keyid, certfile])
+
+
+def run_agent() -> bool:
+    log.info("Running EdgeLock2GO agent...")
+    try:
+        subprocess.check_output(["nxp_iot_agent_demo"], stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        print("| " + e.output.decode().replace("\n", "\n| "), file=sys.stderr)
+        return False
+    log.info("EdgeLock2GO agent check-in complete")
+    return True
+
+
+class AwsKpHandler(NamedTuple):
+    key_id: str = "0X83000044"
+    crt_id: str = "0X83000045"
```

Comment:
How do we know that the key/cert ids will stay the same?

Reply:
These are now managed via our API with constants:

 https://git.foundries.io/development/cloudplatforms/infrastructure/ota-lite/-/blob/master/ota_lite/el2g.py#L13-16

It is a good idea for me to include those constants in our API docs. I'll put a change up for that with my team.

Reply:
Cool, sounds good.